presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDatabaseMetaData.java

```diff
@@ -1440,10 +1442,16 @@ private static void optionalStringInFilter(List<String> filters, String columnNa
         filters.add(filter.toString());
     }
 
-    private static void optionalStringLikeFilter(List<String> filters, String columnName, String value)
+    private void optionalStringLikeFilter(List<String> filters, String columnName, String value)
+            throws SQLException
     {
         if (value != null) {
-            filters.add(stringColumnLike(columnName, value));
+            if (stringIsLikePattern(value)) {
+                filters.add(stringColumnLike(columnName, value));
+            }
+            else {
+                filters.add(stringColumnEquals(columnName, value.replaceAll("\\" + getSearchStringEscape(), "")));
```

Comment:
Indeed. If the intention is to use `getSearchStringEscape()` from `value`, `String.replace` should be used instead.